manta-pay/src/crypto/poseidon/hash.rs

```diff
@@ -134,31 +183,34 @@ where
     }
 }
 
-impl<D, S, const ARITY: usize, COM> Sample<D> for Hasher<S, ARITY, COM>
+impl<D, S, T, const ARITY: usize, COM> Sample<D> for Hasher<S, T, ARITY, COM>
 where
     D: Clone,
     S: Specification<COM>,
     S::Field: Sample<D>,
     S::ParameterField: Field + FieldGeneration + PartialEq + Sample<D>,
+    T: DomainTag<S>,
 {
-    /// Samples random Poseidon parameters.
     #[inline]
     fn sample<R>(distribution: D, rng: &mut R) -> Self
     where
         R: RngCore + ?Sized,
     {
-        // FIXME: Use a proper domain tag sampling method.
-        Self::new(rng.sample(distribution.clone()), rng.sample(distribution))
+        Self::new(rng.sample(distribution))
     }
 }
 
 /// Testing Suite
 #[cfg(test)]
 mod test {
+    use crate::{config::Poseidon2, crypto::constraint::arkworks::Fp};
+    use ark_bls12_381::Fr;
+    use ark_ff::field_new;
+    use manta_crypto::rand::{OsRng, Sample};
+
     /// Tests if [`Poseidon2`](crate::config::Poseidon2) matches the known hash values.
```

Comment:
"known hash values": what hash values?
(I remember we have sage script in `manta-rs` already, maybe add a pointer here?)

Reply:
Added sage for hardcoded tests.